builtin/common/class.lua

```diff
@@ -0,0 +1,119 @@
+class = {}
+
+function class.new(name, ...)
+	local cls = {}
+
+	-- Copy inherited classes in reverse order to make the first ones overwrite later ones
```

Comment:
I would still prefer nested metatables, or possibly a combined approach to allow inheriting metamethods. If chains get long enough to negatively affect performance, that is on the modder (there is a limit on chain length imposed by Lua though).